src/JSTest.Runner/Environment/Node/DebugLogger.ts

```diff
@@ -60,6 +70,10 @@ export class DebugLogger implements IDebugLogger {
     }
 
     public log(message: string) {
-        this.logFileStream.write(message + '\n');
+        if (this.logFileCreated) {
+            this.logFileStream.write(message + '\n');
+        } else {
+            this.messageBuffer.push(message);

```

Comment:
>messageBuffer [](start = 17, length = 13)

when does the message buffer get flushed..?

Reply:
message buffer is for scenarios when the file hasn't been created but logs are being pumped in

```
         this.logFileStream.on('open', () => {
             this.messageBuffer.forEach(message => {
                 this.logFileStream.write(message + '\n');
             });
             this.messageBuffer = [];
             this.logFileCreated = true;
         });
```

in DebugLogger.ts:72 ensures that